integration-tests/deployment/ccip/add_chain.go

```diff
@@ -0,0 +1,170 @@
+package ccipdeployment
+
+import (
+	"math/big"
+
+	"github.com/smartcontractkit/ccip-owner-contracts/tools/proposal/mcms"
+	"github.com/smartcontractkit/ccip-owner-contracts/tools/proposal/timelock"
+	chainsel "github.com/smartcontractkit/chain-selectors"
+
+	"github.com/smartcontractkit/chainlink-ccip/chainconfig"
+	"github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+	"github.com/smartcontractkit/chainlink/integration-tests/deployment"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/ccip_config"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/fee_quoter"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/onramp"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/keystone/generated/capabilities_registry"
+)
+
+// NewChainInboundProposal generates a proposal
+// to connect the new chain to the existing chains.
+func NewChainInboundProposal(
+	e deployment.Environment,
+	state CCIPOnChainState,
+	homeChainSel uint64,
+	newChainSel uint64,
+	sources []uint64,
+) (*timelock.MCMSWithTimelockProposal, error) {
+	// Generate proposal which enables new destination (from test router) on all source chains.
+	var batches []timelock.BatchChainOperation
+	metaDataPerChain := make(map[mcms.ChainIdentifier]timelock.MCMSWithTimelockChainMetadata)
+	for _, source := range sources {
+		chain, _ := chainsel.ChainBySelector(source)
+		enableOnRampDest, err := state.Chains[source].OnRamp.ApplyDestChainConfigUpdates(SimTransactOpts(), []onramp.OnRampDestChainConfigArgs{
```

Comment:
Does this actually "send" the tx to the simchain? Proposals just need calldata right? I'm wondering what'll happen here for the non-sim chains which have "automine" or similar on.

Reply:
it doesn't - uses the `NoSend` flag in transact opts, just a scalable way to get the calldata for proposals

Reply:
Oh that's clever, alright did not notice that